pyro/contrib/gp/util.py

```diff
@@ -128,3 +130,82 @@ def _register_param(self, param, mode="model"):
             p = pyro.sample(param_name, dist.Delta(MAP_param))
 
         self._registered_params[param] = p
+
+
+class LowerConfidenceBound(nn.Module):
```

Comment:
This doesn't need to be a class, you can just split it into stateless functions and move it into the tutorial.

Reply:
Agree, I think it makes more sense to simply provide a collection of functions to do the optimization. Rather than embedding the optimizer into the class.
In practices people have quite different optimization preference like random start, DIRECT etc.

Reply:
I see. The reason I made it a class was to have a universal Acquisition class (with all helper methods) which only requires implementation of `forward` pass for derived classes. However, I agree that it is not necessary for this single tutorial, especially when we decide to put its implementation into the notebook. About @tobyclh points about "in practices, ...", it is one of my most worrisome things with designing API for BO. I am willing to see news with your work! ;)

Reply:
A better way to do this is (I think) is to call this class something like "OptimizingGP" or "BOGP" which takes an acquisition function as an argument. I do like the fact that it is a class though. It should be.

Reply:
@fehiepsi @ysaatchi I think moving everything into functions in the tutorial is fine for now.  I'm writing a design doc for a Pyro Bayesian optimization package where we can discuss what the actual API should look like.  Unfortunately I'm away from work for 3 days starting tomorrow so I'll try to finish adding my thoughts and post it on #836 on Monday for others to comment/edit/extend.